wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession+UserSession.swift

```diff
@@ -294,6 +294,31 @@ extension ZMUserSession: UserSession {
         let featureRepository = FeatureRepository(context: syncContext)
         return GetMLSFeatureUseCase(featureRepository: featureRepository)
     }
+
+    @MainActor
+    public func fetchSelfConversationMLSGroupID() async -> MLSGroupID? {
+        return await syncContext.perform {
+            return ZMConversation.fetchSelfMLSConversation(in: self.syncContext)?.mlsGroupID
+        }
+    }
+
+    @MainActor
+    public func e2eIdentityUpdateCertificateUpdateStatus() async -> E2EIdentityCertificateUpdateStatusUseCaseProtocol? {
+        guard let selfUserClient,
+              let mlsGroupID = await self.fetchSelfConversationMLSGroupID(),
```

Comment:
If we always need the id of the self-mls-conversation, there is no point in injecting it into the use case.
It should rather be part of the use case to fetch the conversation.

Then the `async` can be removed from this method.

Reply:
done